package.json

```diff
@@ -100,7 +100,8 @@
     "pino-pretty": "^7.6.1",
     "pnpm": "^10.0.0",
     "prom-client": "^14.0.1",
-    "redis": "^4.7.0"
+    "redis": "^4.7.0",
+    "typed-emitter": "^2.1.0"
```

Comment:
It looks like this package `typed-emitter` is quite old (and probably unmaintained).

Maybe it's better to use Node's native typings. We are using an old version of `@types/node` (`17.x.x`) which doesn't include support for typed events. However, a newer version https://www.npmjs.com/package/@types/node/v/20.14.8?activeTab=code does include typed events (see `events.d.ts` for event definitions).